### PR TITLE
metrics: add replication metrics on proxied requests

### PR DESCRIPTION
**Pull request by poornas**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
dependency https://github.com/minio/madmin-go/pull/268

## How to test this PR?
set up bucket/site replication and see prometheus metrics for `mc stat` or `mc head` on non-existent object/version - the metrics count should reflect proxied requests.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Review by harshavardhana — APPROVED** [review]

File: docs/metrics/prometheus/list.md [hunk 1]
```diff
@@ -290,6 +296,11 @@ For deployments with [Site Replication](https://min.io/docs/minio/linux/operatio
 | `minio_bucket_replication_sent_bytes`               | Total number of bytes replicated to the target bucket.                           |
 | `minio_bucket_replication_sent_count`               | Total number of objects replicated to the target bucket.                         |
 | `minio_bucket_replication_credential_errors`        | Total number of replication credential errors since server start                 |
+| `minio_bucket_replication_proxied_get_requests_count` | Number of GET requests proxied to replication target                          |
```
Comment (harshavardhana) [inline docs/metrics/prometheus/list.md]:
```suggestion
| `minio_bucket_replication_proxied_get_requests_total` | Number of GET requests proxied to replication target                          |
```

File: docs/metrics/prometheus/list.md [hunk 2]
```diff
@@ -290,6 +296,11 @@ For deployments with [Site Replication](https://min.io/docs/minio/linux/operatio
...
+| `minio_bucket_replication_proxied_head_requests_count` | Number of HEAD requests proxied to replication target                          |
```
Comment (harshavardhana) [inline docs/metrics/prometheus/list.md]:
```suggestion
| `minio_bucket_replication_proxied_head_requests_total` | Number of HEAD requests proxied to replication target                          |
```

File: docs/metrics/prometheus/list.md [hunk 3]
```diff
@@ -290,6 +296,11 @@ For deployments with [Site Replication](https://min.io/docs/minio/linux/operatio
...
+| `minio_bucket_replication_proxied_delete_tagging_requests_count` | Number of DELETE requests proxied to replication target                          |
```
Comment (harshavardhana) [inline docs/metrics/prometheus/list.md]:
```suggestion
| `minio_bucket_replication_proxied_delete_tagging_requests_total` | Number of DELETE requests proxied to replication target                          |
```

File: docs/metrics/prometheus/list.md [hunk 4]
```diff
@@ -290,6 +296,11 @@ For deployments with [Site Replication](https://min.io/docs/minio/linux/operatio
...
+| `minio_bucket_replication_proxied_get_tagging_requests_count` | Number of GET tagging requests proxied to replication target                          |
```
Comment (harshavardhana) [inline docs/metrics/prometheus/list.md]:
```suggestion
| `minio_bucket_replication_proxied_get_tagging_requests_total` | Number of GET tagging requests proxied to replication target                          |
```

File: docs/metrics/prometheus/list.md [hunk 5]
```diff
@@ -290,6 +296,11 @@ For deployments with [Site Replication](https://min.io/docs/minio/linux/operatio
...
+| `minio_bucket_replication_proxied_put_tagging_requests_count` | Number of PUT tagging requests proxied to replication target                          |
```
Comment (harshavardhana) [inline docs/metrics/prometheus/list.md]:
```suggestion
| `minio_bucket_replication_proxied_put_tagging_requests_total` | Number of PUT tagging requests proxied to replication target                          |
```

Comment (harshavardhana) [reply docs/metrics/prometheus/list.md]: We should also capture the proxies that failed and if the failed is increasing or not.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: docs/metrics/prometheus/list.md (hunk 5 above)
Comment (poornas) [reply docs/metrics/prometheus/list.md]: done

File: cmd/metrics-v2.go [hunk 6]
```diff
@@ -225,10 +225,20 @@ const (
...
+	maxInQueueBytes        MetricName = "max_queued_bytes"
+	proxiedGetReqTotal     MetricName = "proxied_get_requests_total"
+	proxiedHeadReqTotal    MetricName = "proxied_head_requests_total"
+	proxiedPutTagReqTotal  MetricName = "proxied_put_tagging_requests_total"
+	proxiedGetTagReqTotal  MetricName = "proxied_get_tagging_requests_total"
+	proxiedRmvTagReqTotal  MetricName = "proxied_delete_tagging_requests_total"
+	proxiedGetReqFailed    MetricName = "proxied_get_requests_failures"
+	proxiedHeadReqFailed   MetricName = "proxied_head_requests_failures"
+	proxiedPutTagReqFailed MetricName = "proxied_put_tagging_requests_failures"
+	proxiedGetTagReqFailed MetricName = "proxied_get_tagging_requests_failures"
```
Comment (harshavardhana) [inline cmd/metrics-v2.go]:
```suggestion
	proxiedGetReqFailed    MetricName = "proxied_get_requests_failed"
	proxiedHeadReqFailed   MetricName = "proxied_head_requests_failed"
	proxiedPutTagReqFailed MetricName = "proxied_put_tagging_requests_failed"
	proxiedGetTagReqFailed MetricName = "proxied_get_tagging_requests_failed"
```

Match the name of the metric

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: docs/metrics/prometheus/list.md [hunk 7]
```diff
@@ -290,6 +301,16 @@ For deployments with [Site Replication](https://min.io/docs/minio/linux/operatio
 | `minio_bucket_replication_sent_bytes`               | Total number of bytes replicated to the target bucket.                           |
 | `minio_bucket_replication_sent_count`               | Total number of objects replicated to the target bucket.                         |
 | `minio_bucket_replication_credential_errors`        | Total number of replication credential errors since server start                 |
+| `minio_bucket_replication_proxied_get_requests_count` | Number of GET requests proxied to replication target                          |
```
Comment (harshavardhana) [inline docs/metrics/prometheus/list.md]: This list still says _count

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by kannappanr — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**